### PR TITLE
gprojector: init at 3.0.2

### PR DESCRIPTION
**Pull request by cyclic-pentane**
###### Motivation for this change
[G.Projector](https://www.giss.nasa.gov/tools/gprojector/) is a program written by the NASA GISS to deal with map projections (e. g. for satellite images or topology maps). Marked as "nonfree" because the homepage doesn't mention a license nor provides any source code.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ stdenv
+, lib
+, fetchzip
+, jre
+, makeDesktopItem
+, makeWrapper
+, ...
```
Comment (bobby285271) [inline pkgs/applications/science/astronomy/gprojector/default.nix:7]:
```suggestion
```

https://nixos.org/manual/nixpkgs/stable/#sec-syntax

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    desktopName = "G.Projector";
+    comment = meta.description;
+    categories = "Science;";
+    extraEntries = "StartupWMClass = gov-nasa-giss-projector-GProjector";
+  };
+
+  buildInputs = [ jre ];
+  nativeBuildInputs = [ makeWrapper desktopItem ];
+
+  phases = [ "installPhase" ];
```
Comment (bobby285271) [inline pkgs/applications/science/astronomy/gprojector/default.nix:31]: We are deprecating explicit setting of `phases`, see #28910.

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    runHook preInstall
+    mkdir -p $out/share
+    cp -r $src/jars $out/share/java
+    cp -r $desktopItem/share/applications $out/share/applications
+    makeWrapper ${jre}/bin/java $out/bin/gprojector --add-flags "-Xms512M -Xmx2000M -jar $out/share/java/G.Projector.jar"
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "G.Projector transforms an input map image into any of about 200 global and regional map projections.";
```
Comment (bobby285271) [inline pkgs/applications/science/astronomy/gprojector/default.nix:43]: As [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) suggested `meta.description` shouldn't have a period at the end.

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    name = "gprojector";
+    exec = "gprojector";
+    desktopName = "G.Projector";
+    comment = meta.description;
+    categories = "Science;";
+    extraEntries = "StartupWMClass = gov-nasa-giss-projector-GProjector";
+  };
+
+  buildInputs = [ jre ];
+  nativeBuildInputs = [ makeWrapper desktopItem ];
```
Comment (sikmir) [inline pkgs/applications/science/astronomy/gprojector/default.nix:28]:
```suggestion
  nativeBuildInputs = [ makeWrapper copyDesktopItems ];
```

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "gprojector";
+    exec = "gprojector";
+    desktopName = "G.Projector";
+    comment = meta.description;
+    categories = "Science;";
+    extraEntries = "StartupWMClass = gov-nasa-giss-projector-GProjector";
+  };
```
Comment (sikmir) [inline pkgs/applications/science/astronomy/gprojector/default.nix:25]:
```suggestion
  desktopItems = [
    (makeDesktopItem {
      name = "gprojector";
      exec = "gprojector";
      desktopName = "G.Projector";
      comment = meta.description;
      categories = "Science;";
      extraEntries = "StartupWMClass = gov-nasa-giss-projector-GProjector";
    })
  ];
```

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+  dontConfigure = true;
+  dontBuild = true;
+  dontFixup = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share
+    cp -r $src/jars $out/share/java
+    cp -r $desktopItem/share/applications $out/share/applications
```
Comment (sikmir) [inline pkgs/applications/science/astronomy/gprojector/default.nix:38]:
```suggestion
```

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{ stdenv
+, lib
+, fetchzip
+, jre
+, makeDesktopItem
```
Comment (sikmir) [inline pkgs/applications/science/astronomy/gprojector/default.nix:5]:
```suggestion
, makeDesktopItem
, copyDesktopItems
```

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  dontConfigure = true;
+  dontBuild = true;
+  dontFixup = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share
+    cp -r $src/jars $out/share/java
+    cp -r $desktopItem/share/applications $out/share/applications
+    makeWrapper ${jre}/bin/java $out/bin/gprojector --add-flags "-Xms512M -Xmx2000M -jar $out/share/java/G.Projector.jar"
```
Comment (sikmir) [inline pkgs/applications/science/astronomy/gprojector/default.nix:39]: Maybe top level parameter:
`extraJavaOpts ? "-Xms512M -Xmx2000M"`
to make it possible to override if needed.

File: pkgs/applications/science/astronomy/gprojector/default.nix (hunk 4 above)
Comment (alyaeanyx) [reply pkgs/applications/science/astronomy/gprojector/default.nix]: I didn't know about copyDesktopItems for far, thanks for the suggestion :)

File: pkgs/applications/science/astronomy/gprojector/default.nix [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+, lib
+, fetchzip
+, jre
+, makeDesktopItem
+, copyDesktopItems
+, makeWrapper
+, extraJavaArgs ? "-Xms512M -Xmx2000M"
+}:
+
+stdenv.mkDerivation rec {
```
Comment (06kellyjac) [inline pkgs/applications/science/astronomy/gprojector/default.nix:11]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

If this is just wrapping gprojector can this use `stdenvNoCC`?

Comment (alyaeanyx) [reply pkgs/applications/science/astronomy/gprojector/default.nix]: Yep, you're absolutely right, thanks for the suggestion!

**Pull request merged**